handlers/deployment.rb

```diff
@@ -189,6 +205,62 @@ def jenkins
                                             ssl: true)
       end
 
+      def get_repo_status(repo_name)
+        deployed_version = get_deployed_commit(repo_name)
+        production_tag = production_release_tag(repo_name)
+        get_app_status(repo_name, deployed_version, production_tag)
+      end
+
+      def get_app_status(repo_name, deployed_version, prod_tag)
+        git_responses = {}
+        ['HEAD', prod_tag].each do |tag|
+          comparison = Octokit.compare(repo_name, deployed_version.strip, tag)
+          if comparison.commits.empty?
+            git_responses[tag] = 'is the currently deployed version.'
+          else
+            word = comparison.commits.size > 1 ? 'commits' : 'commit'
+            git_responses[tag] = "#{comparison.commits.size} undeployed #{word}. #{comparison.permalink_url}"
+            git_responses[tag] << ' :shipit:' if tag == 'production-release'
+          end
+        end
+
+        formatted_response = git_responses.map do |tag, comment|
+          "#{tag.upcase} : #{comment}"
+        end
+
+        formatted_response.join("\n")
+      end
+
+      def get_deployed_commit(repo_name)
+        type, repo_url = repo_type_and_url(repo_name)
+        repo_url_data = HTTParty.get(repo_url)
+        case type
+        when 'json'
+          commit_key = (JSON_COMMIT_ID_KEYS & repo_url_data.keys).first
+          repo_url_data.fetch(commit_key)
+        else
+          repo_url_data
+        end
+      end
+
+      def repo_type_and_url(repo_name)
+        type, url = ORG_URL_AND_TYPES[repo_name]
+        if type && url
+          [type, url]
+        else
+          app_name = repo_name.split('/')[1]
+          ['file', "https://#{app_name}.zooniverse.org/commit_id.txt"]
```

Comment:
I'm fine with us merging this as it is, but I do want to point out that this assumes the app is deployed under the same name as its repo. That's a sensible naming convention for us to use, so we should just make sure we enforce that when we deploy new apps.

Reply:
yes - if it's not it'll be in the exceptions list defined in https://github.com/zooniverse/lita/blob/192d39dc1b661d7c80b6ebde3a37665292defaa2/handlers/deployment.rb#L28

This works for now, i think we standardize on a file / json resposne for status on our app services which includes the deployed commit_id.